docs/user-guide/sync-options.md

```diff
@@ -227,4 +227,37 @@ spec:
     syncOptions:
     - CreateNamespace=true
 ```
-The example above shows how an Argo CD Application can be configured so it will create namespaces for the Application resources if the namespaces don't exist already. Without this either declared in the Application manifest or passed in the cli via `--sync-option CreateNamespace=true`, the Application will fail to sync if the resources' namespaces do not exist.
+The example above shows how an Argo CD Application can be configured, so it will create namespaces for the Application resources if the namespaces don't exist already. Without this either declared in the Application manifest or passed in the cli via `--sync-option CreateNamespace=true`, the Application will fail to sync if the resources' namespaces do not exist.
+
+### Namespace Metadata
+
+We can also add labels and annotations to the namespace through `managedNamespaceMetadata`. If we extend example above 
+we could potentially do something like below:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  namespace: test
+spec:
+  syncPolicy:
+    managedNamespaceMetadata:
+      labels: # The labels to set on the application namespace
+        any: label
+        you: like
+      annotations: # The annotations to set on the application namespace
+        the: same
+        applies: for
+        annotations: on-the-namespace
+    syncOptions:
+    - CreateNamespace=true
+```
+
+There are a few things to keep in mind when using `managedNamespaceMetadata`.
+
+* Setting `CreateNamespace=true` is a prerequisite, otherwise nothing will happen.
+* In case the namespace already exists, ArgoCD will take over the management of said namespace and set its labels and annotations as per the contents in `managedNamespaceMetadata`.
+* If the application has a namespace manifest which has conflicting definitions of labels and/or annotations, that takes precedence over what is specified in `managedNamespaceMetadata`, and will *overwrite* the contents of `managedNamespaceMetadata`.
+
+Using `managedNamespaceMetadata` will also set the resource tracking label (or annotation) on the namespace, so you can
```

Comment:
I haven't yet implemented the addition of `resourceTracking` on the namespace. Does this make sense to implement @crenshaw-dev?

Reply:
I think it would... we'd add resource tracking if the user added the namespace via manifests, and I think we want this to closely mirror that experience.